homeassistant/components/media_player/__init__.py

```diff
@@ -866,6 +866,14 @@ def preload_media_image_url(self, url):
             _async_fetch_image(self.hass, url), self.hass.loop
         ).result()
 
+    def set_last_bad_image_url(self, url):
+        """Save last bad image url."""
+        should_update = self._last_bad_image_url != url
+        self._last_bad_image_url = url
+        if should_update:
+            _LOGGER.debug('%s marked as inaccessible', url)
+            self.schedule_update_ha_state()
```

Comment:
`schedule_update_ha_state` is only to be called from a threaded context. It should not be added from an async context.

Also, I don't see a reason to update the entity because the image fetch failed, because it means it will trigger another image fetch in the frontend?